AP-protocol.md

```diff
@@ -66,33 +66,70 @@ Recall that there are: Content servers (that is, Text and Media servers) and Com
 
 These servers exchange, respectively, Text server messages, Media server messages and Communication server messages. These are high-level messages. Recall that you must standardize and regulate their low-level counterparts (that is, fragments).
 
-# Source routing
+# Source Routing Protocol
 
 The fragments that circulate in the network are **source-routed** (except for the commands sent from and the events received by the Simulation Controller).
 
-Source routing refers to a technique where the sender of a data packet specifies the route the packet takes through the network. This is in contrast with conventional routing, where routers in the network determine the path incrementally based on the packet's destination.
+Source routing refers to a technique where the sender of a data packet specifies the route the packet takes through the network. This is in contrast with conventional routing, where routers in the network determine the path incrementally based on the packet's destination.
 
 The consequence is that drones do not need to maintain routing tables.
 
-As an example, consider the following simplified network:
+### How Source Routing Works
+
+When a client or server wants to send a message to another node, it performs the following steps:
+
+1. **Route Computation**: The sender calculates the entire path to the destination node. This path includes the sender itself and all intermediate nodes leading to the destination.
+
+2. **Creation of the Source Routing Header**: The sender constructs a header that contains:
+	- **`hops`**: A list of node IDs representing the route from the sender to the destination.
+	- **`hop_index`**: An index indicating the current position in the `hops` list. It starts at **1** because the first hop (`hops[0]`) is the sender itself.
+
+3. **Packet Sending**: The sender attaches the source routing header to the packet and sends it to the first node in the route (the node at `hops[1]`).
+
+### Step-by-Step Example
+
+Consider the following simplified network topology:
 
 ![constellation](assets/costellation.png)
 
-Suppose that the client A wants to send a message to the server D.
+Suppose that client A wants to send a message to server D.
+
+**Client A**:
+
+- Computes the route: **A → B → E → F → D**.
+- Creates a source routing header:
+	- **`hops`**: `[A, B, E, F, D]`.
+	- **`hop_index`**: `1`.
+- Sends the packet to **B**, the first node after itself.
+
+**At Each Node**:
 
-It computes the route B→E→F→D, creates a **Source Routing Header** specifying route A→B→E→F→D, adds it to the packet and sends it to B.
+1. **Node B**:
+	- Receives the packet.
+	- Determines that the next hop is **E**.
+	- Sends the packet to **E**.
```

Comment:
```suggestion
1. **Node B**:
	- Receives the packet.
	- Increments the **`hop_index`** by 1.
	- Determines that the next hop is **E**.
	- Sends the packet to **E**.
```
This needs to be applied also to the following steps.

Reply:
yes, it was to not shift the focus to the Drone Protocol but to leave the focus of the example on the Source Routing Protocol. I will now however move everything to an example file as suggested by @matteo-dinoia

Reply:
@DiCesareDaniele is right, better to move the examples in a separate PR